jfrog-cli/artifactory/utils/golang/project/dependencies/packageswithdeps.go

```diff
@@ -0,0 +1,256 @@
+package dependencies
+
+import (
+	"fmt"
+	golangutil "github.com/jfrog/jfrog-cli-go/jfrog-cli/artifactory/utils/golang"
+	"github.com/jfrog/jfrog-cli-go/jfrog-cli/utils/config"
+	"github.com/jfrog/jfrog-client-go/httpclient"
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+	"github.com/jfrog/jfrog-client-go/utils/io/fileutils"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+// Represent go dependency package when running with deps tidy set to true.
+type PackageWithDeps struct {
+	Dependency             *Package
+	transitiveDependencies []PackageWithDeps
+	regExp                 *RegExp
+	runGoModCommand        bool
+	tempDir                string
+	cachePath              string
+}
+
+// Creates a new dependency
+func (pwd *PackageWithDeps) New(cachePath string, dependency Package) GoPackage {
+	pwd.Dependency = &dependency
+	pwd.cachePath = cachePath
+	pwd.transitiveDependencies = nil
+	return pwd
+}
+
+// Performs population of the mod file and publish the dependency and it's transitive dependencies to Artifactory
```

Comment:
// Performs population of the ... --> // Populate the ...

Reply:
Done